src/services/fileStorage/proxy-service.js

```diff
@@ -473,6 +473,10 @@ const directoryService = {
 						.catch(() => undefined),
 				);
 				return Promise.all(permissionPromises);
+			})
+			.then((allowedFiles) => {
+				const files = allowedFiles.filter(f => f);
+				return files.length ? files : new NotFound();
```

Comment:
It can remove Arrays without elements, but if no array then return NotFound() ? 
Can you also pass a short information to NotFound ?

Reply:
`files` is always an array at this point. @CeEv what kind of information do you mean?